### PR TITLE
Add humidex indicator

### LINKED ISSUE #665: Humidex indicator
### Description

Add humidex indicator. 

https://www.canada.ca/en/environment-climate-change/services/seasonal-weather-hazards/warm-season-weather-hazards.html#toc7

https://en.wikipedia.org/wiki/Humidex

### PR DESCRIPTION
**Pull request by huard**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #665
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [ ] `bumpversion (major / minor / patch)` has been called on this branch
- [ ] Tags have been pushed (`git push --tags`)

* **What kind of change does this PR introduce?** 
Add humidex indicator

* **Does this PR introduce a breaking change?** 
No

* **Other information**:

Questions: 
- Should it return an integer ?
- Should it support `hurs` as an input (in addition to the dewpoint temperature) ?

### CODE REVIEW

File: xclim/data/fr.json [hunk 1]
```diff
@@ -832,5 +832,11 @@
     "abstract": "Indice basé sur la température utilisé pour estimer le développement des cultures de maïs. La croissance du maïs se produit lorsque la température journalière minimale et maximale excèdent des seuils donnés.",
     "description": "Indice basé sur la température utilisé pour estimer le développement des cultures de maïs. La croissance du maïs se produit lorsque la température journalière minimale et maximale excèdent {thresh_tasmin} et {thresh_tasmax}, respectivement.)",
     "long_name": "Unités thermiques maïs (Tmin > {thresh_tasmin} et Tmax > {thresh_tasmax})."
+  },
+    "HUMIDEX": {
+    "title": "Indice humidex",
+    "abstract": "L'indice humidex décrit la température ressentie par une personne lorsqu'on tient compte de l'humidité relative. Il peut être interpreté comme la température équivalente ressentie lorsque l'air est sec.",
+    "description": "Indice humidex décrivant la température ressentie par une personne en réponse à l'humidité relative.",
+    "long_name": ""
```
Comment (aulemahal) [inline xclim/data/fr.json:840]:
```suggestion
    "long_name": "Indice humidex"
```
Non?

**Review by aulemahal — COMMENTED** [review]
J'imagine que c'est bon d'ajouter une option avec `hurs`? Je sais pas qui va utiliser cet index-là, mais il me semble que la température du point de rosée n'est pas non plus si courante dans les modèles? Ça peut être juste pour l'indice si ça fait trop complexe.

File: xclim/data/fr.json (hunk 1 above)
Comment (huard) [reply xclim/data/fr.json]: oui !

**Comment by huard** [comment]
My concern was that this is something already in [MetPy](https://unidata.github.io/MetPy/latest/api/generated/metpy.calc.dewpoint_from_relative_humidity.html#dewpoint-from-relative-humidity), so not sure if we want to start to mirror this functionality here, or rather solve #703 ?

**Comment by aulemahal** [comment]
I can take a few minutes today to explore #703 in more depth!

File: HISTORY.rst [hunk 2]
```diff
@@ -34,9 +34,13 @@ New indicators
...
+* `atmos.humidex` returns the Canadian *humidex*, an indicator of perceived temperature account for relative humidity.
 * `multiday_temperature_swing` indice for returning general statistics based on spells of doubly-thresholded temperatures (Tmin < T1, Tmax > T2).
 * New indicators `atmos.freezethaw_frequency`, `atmos.freezethaw_spell_mean_length`, `atmos.freezethaw_spell_max_length` for statistics of Tmin < 0 degC and Tmax > 0 deg C days now available (wrapped from `multiday_temperature_swing`).
 * `atmos.wind_chill_index` computes the daily wind chill index. The default is similar to what Environment and Climate Change Canada does, options are tunable to get the version of the National Weather Service.
+* `multiday_temperature_swing` indice for returning general statistics based on spells of doubly-thresholded temperatures (Tmin < T1, Tmax > T2).
+* New indicators `atmos.freezethaw_frequency`, `atmos.freezethaw_spell_mean_length`, `atmos.freezethaw_spell_max_length` for statistics of Tmin < 0 degC and Tmax > 0 deg C days now available (wrapped from `multiday_temperature_swing`).
+
```
Comment (aulemahal) [inline HISTORY.rst:43]:
```suggestion
```
Haha sorry, still some leftovers! (See above)

**Review by aulemahal — APPROVED** [review]

**Pull request merged**